### ISSUE TITLE
Question about Contexts & Hooks

### ISSUE THREAD

**Issue by r002** [kickstartcoding/reactjs.org#3]
Thank you, @neurodynamic, for putting this site together.  Loving React so much more since the introduction of hooks!  I've been learning more about context and hooks and specifically, on:

https://reactwithhooks.netlify.app/docs/context.html#updating-context-from-a-nested-component,

![image](https://user-images.githubusercontent.com/45280066/111480702-a6089300-8708-11eb-86d3-d30675ab0070.png)

is this the only way to pass the `toggleTheme` function down to all children?  To set up another context?  It just seems having to set up **_multiple_** providers (one for _being_ the theme and one for _changing_ the theme) feels awfully clunky.

Is there any better way to just put all theme-related functionality (defining and toggling) into a single context?

PS. A big thanks to [armenic](https://github.com/reactjs/reactjs.org/issues/3308#issuecomment-755744743).  That's how I found [this website](https://codesandbox.io/s/react-hooks-update-context-from-nested-component-50y5j)!

**Comment by neurodynamic** [comment]
@r002 Hey, thanks; glad you like it!

It was the only way I could figure out to do it, and I agree it feels very clunky. There may be a better way, but I wasn't able to figure it out when I was building this. The core team might have a better idea?

If you find out, definitely let me know so I can update the examples.

**Comment by r002** [comment]
Hi @neurodynamic !  Just a quick update.  Today I spent some time and finally figured this out.  I studied the way it was done [in old React class components](https://reactjs.org/docs/context.html#updating-context-from-a-nested-component) and then just extended the logic to hooks.  I forked [armenic's original](https://codesandbox.io/s/react-hooks-update-context-from-nested-component-50y5j) and quickly coded [this working example](https://codesandbox.io/s/react-hooks-update-context-from-nested-component-forked-t70up?file=/src/App.js).

If you like this, lmk and I can submit a PR to include these changes in your [**"React with Hooks"** documentation](https://reactwithhooks.netlify.app/docs/context.html#updating-context-from-a-nested-component).  Thanks again for putting your guide together; it's been really helpful! 🙏

Here are the highlights:

### `App.js`
```js
export const App = () => {
  const [themeContext, setThemeContext] = useState({
    theme: themes.light,
    toggler: customThemeToggler
  })

  function customThemeToggler() {
    setThemeContext((oldContext) => ({
      theme: oldContext.theme === themes.light ? themes.dark : themes.light,
      toggler: customThemeToggler
    }))
  }
...
```

### `theme-context.js`
```js
// Default ThemeContext values.
// This will never appear if overridden in /App.js
export const ThemeContext = React.createContext({
  theme: themes.dark,
  toggler: () => {
    console.log(
      "Default toggler has been called! " +
        "If I am overriden, I will never appear!"
    )
  }
})
```

### `theme-toggler-button.js`
```js
  ...
  const themeContext = useContext(ThemeContext)

  return (
    <button
      onClick={themeContext.toggler}
      style={{ backgroundColor: themeContext.theme.background }}
    >
      Toggle Theme
    </button>
  )
```

**Comment by neurodynamic** [comment]
@r002 Thank you for this! A PR would be great!

**Comment by r002** [comment]
You got it! I'll get this done and submit a PR for it tomorrow! 👍

**Comment by r002** [comment]
Just a quick update:  Hit a few snags last night and everything took me a little longer to set up than expected.  Finally figured everything out and managed to get your repo built and running locally.  The problem happened because I am using Windows and so when I cloned and tried to run `yarn dev` on your fork, I got GraphQL errors, right out of the box:

![image](https://user-images.githubusercontent.com/45280066/113479224-c734e780-945b-11eb-8422-a63e9de884cd.png)

The problem isn't unique to your fork though.  The original `reactjs.org` repo [had the same problem in June 2020](https://github.com/reactjs/reactjs.org/issues/3012) which was patched in [this August 2020 PR](https://github.com/reactjs/reactjs.org/pull/3213).  The problem only happens in Windows because [Gatsby uses `path.join` with forward-slash](https://github.com/reactjs/reactjs.org/pull/3213#issue-470204481) which is fine on Macs but will break Windows.

Anyway, just wanted to let you know in case you wanted to grab that specific PR from `react.js` trunk.

I've begun writing the updated `context with hooks` documentation-- will submit the PR later tonight.
